pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -121,6 +120,9 @@ func NewFramework(r Registry, plugins *config.Plugins, args []config.PluginConfi
 					return nil, fmt.Errorf("plugin %v does not extend score plugin", sc.Name)
 				}
 				f.scorePlugins = append(f.scorePlugins, p)
+				if _, exists := f.pluginNameToWeightMap[p.Name()]; !exists {
```

Comment:
If the plugin could be found from `pluginsMape` at L117, it could also be found from `pluginNameToWeightMap` because they are initialized at the same block(L92 and L96). Then we could also skip this check.